src/Crud/Controller/GetAll.php

```diff
@@ -62,6 +62,11 @@ public function __invoke(Request $request)
 
         $page = $request->query->getInt('page', 1);
         $maxPerPage = $request->attributes->get(self::MAX_PER_PAGE_ATTRIBUTE, 30);
+        if (true === $request->attributes->getBoolean(self::TAKE_USER_MAX_PAGE, false)) {
+            if (0 !== $userMax = $request->query->getInt('size', 0)) {
```

Comment:
`item_per_page` would be a better name. Having it as const would be also better. Finally, we need some more configuration considering we can't allow the user to get the whole database. We also need a `maximum_items_per_page` config (that could be 300 by default).